src/liballoc/collections/btree/set.rs

```diff
@@ -326,9 +342,51 @@ impl<T: Ord> BTreeSet<T> {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn intersection<'a>(&'a self, other: &'a BTreeSet<T>) -> Intersection<'a, T> {
+        let (a_set, b_set) = if self.len() <= other.len() {
+            (self, other)
+        } else {
+            (other, self)
+        };
+        if a_set.len() > b_set.len() / 16 {
```

Comment:
Might be worth leaving a comment here about what this branch is for and why we use the constant `16` to decide which strategy to use.

Reply:
Comment left, but it seems you have to find it yourself.